### ISSUE TITLE
Bare minimum

### ISSUE THREAD

**Issue by patricksavalle** [mobbr/mobbr-browser-plugin#1]
Mobbr use a metadata tag in the HTML of pages to distribute payments to that URL. We need a browser plugin that can do these payments and show some data on the URL. For instance, The pages on this website have Mobbr metadata, look as the HTML source: https://fastmovingtargets.nl/workforce/653/odeon-de-spiegel-hoe-ziet-het-theater-van-de-toekomst-eruit

The metadata specification is here: https://github.com/mobbr/mobbr-api-v1/tree/master/examples#preparing-a-webpage

Let's start with the absolute minimum: a Chrome extension. See: https://developer.chrome.com/extensions/getstarted

Steps to take:
- Create a Github-project
- Use an example or other plugin as starting point
- The plugin should display a Mobbr button in the browser
- The plugin detects the Mobbr script on the current URL, if present change color of plugin icon / show other color icon
- When clicked, go to the Mobbr page for that URL, which is https://mobbr.com/#/task/<base64_encoded_url> (check here: https://www.base64decode.org/)
- attach the resulting code/project as a pull request to this issue

**Comment by gathumartins** [comment]
Hello Patrick, Am Martin from Nairobits when i click on the first url on this assignment it opens a page written in Dutch.

**Comment by espruyt** [comment]
It doesn't matter. Just check the HTML source (http://www.wikihow.com/View-Source-Code), then you'll find the metadata.

Like in this screenshot:
![schermafbeelding 2015-04-30 om 13 48 16](https://cloud.githubusercontent.com/assets/7690396/7412106/a872db12-ef3f-11e4-9f72-9ea7b122db22.png)

**Comment by gathumartins** [comment]
Cool got it.

On Thu, Apr 30, 2015 at 2:49 PM, espruyt notifications@github.com wrote:

> It doesn't matter. Just check the HTML source, then you'll find the
> metadata.
> 
> Like in this screenshot:
> [image: schermafbeelding 2015-04-30 om 13 48 16]
> https://cloud.githubusercontent.com/assets/7690396/7412106/a872db12-ef3f-11e4-9f72-9ea7b122db22.png
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/mobbr/mobbr-browser-plugin/issues/1#issuecomment-97747326
> .

## 

Martin Gathu Manyeki

Trainer
NairoBits Digital Design School
Leveraging on technology to transform lives

t   + 254 020 557 635
m   +254 728 429 252
e   martin@nairobits.com wilson@nairobits.com
w   www.nairobits.com
a   nairobi godown arts centre, south b

Twitter: www.twitter.com/nairobits
Blog: www.nairobits.com/blog

**Comment by jrjackso** [comment]
Hey Patrick, I don't think you can submit a pull request for an empty repository.
The code/project is here: https://github.com/jrjackso/mobbr-browser-plugin
Would obviously appreciate the credit for the task on mobbr when you copy it in - not sure how mobbr handles that b/c it will look like you created it?

(It's implemented as a page action so when you're on a page such as the example you gave, a mobbr icon will show up in the address bar. If you click that icon it will let you go to the task on mobbr as requested. Page action seemed to make more sense reading the chrome extension docs because this icon/click applies only to pages with participation on it.)

PS. There is a pull request already for another issue you marked high priority if you get a chance to test/merge that too.

**Comment by patricksavalle** [comment]
@jrjackso cool :) I will ask here how we can fix this! I should probably add some minimum scaffolding code, upon which you can (re-) submit a pull request or something 

@gathumartins Hold your horses, please wait until this code is pulled into the repo :) I will add more functionality, more pledges. After all we need a fully functional plugin.

@handijk can you please help me out here? What can we do to allow the pull request from @jrjackso ?

**Comment by jrjackso** [comment]
@patricksavalle I think if you just start the repository with a README.md I can fork/pull request that. Mobbr will then pick me up as a programmer on the issue (and others as commenters)

That said, the other thing that's a bit funky is if you merge my pull request, you also get marked as a programmer for that action (equal weight I think) and take approx. half the bounty :(
If you want an example of this, refer to this task: aHR0cHM6Ly9naXRodWIuY29tL21vYmJyL21vYmJyLWZyb250ZW5kL2lzc3Vlcy8yMDg=

**Comment by jrjackso** [comment]
I'm a bit surprised mobbr is currently showing 0 participation shares for this task actually. Doesn't it normally pick up commenters, issue owners, ...  Has something changed on your end in the scoring?

**Comment by patricksavalle** [comment]
@jrjackso I will look into the sharing. I noticed this too. We have to tweak the algorithm a little. For repos it works fine, for issues it is overenthusiastic for issues owners, closers, etc.

**Comment by gathumartins** [comment]
Hello Patrick. Kindly check for me this project and let me know if I am progressing in the right direction here is the link. https://github.com/gathumartins/marto.git
I will appreciate your comments please.

**Comment by jrjackso** [comment]
@gathumartins for the full-out extension, it might be worth bringing angularjs in (ie: ng-csp). That would mostly mean porting over your static html ( e.g. https://github.com/gathumartins/marto/blob/master/receivers.html ) to angular views, but it would be pretty close css style-wise. That said, if we could pull angularjs in, could more of lightbox itself (views, scripts, styles, directives, ...) be used?

**Comment by handijk** [comment]
@jrjackso i have imported your repo into this one, it still shows your original commits. If you could base your future work on a fork of this repo we are fine.
@patricksavalle i think the participation share for a merge should be based on the complexity of the pull request, a small percentage of what the programmer gets. Issue comments can be worthless, or worth a lot, to bad github issues have no +/- or some kind of rating.
I think if the functionality of the plugin is very close to that of the lightbox and angular is able to run in plugin environments (firefox, ie also) we should share the codebase of the lightbox.
It will save a lot of work when changing basic functionality and the modularity of angular could still allow for added lightbox/plugin functionality.

**Comment by jrjackso** [comment]
Thanks Han. I'll fork the imported repo.
The only thing that comes to mind offhand for comments would be weighting them based on the commenter's overall contribution/participation in the project. (Basically making the assumption that users who have contributed lots would make comments that are more useful and answer questions to unblock people. They're not likely to spam too b/c it's their project.)

**Comment by patricksavalle** [comment]
I will tweak the algorithm and then trigger the pledge.

**Comment by patricksavalle** [comment]
@jrjackso We paid on the repo (instead of this issue). Same result :)

@gathumartins There will be new specs and issues. Stay tuned. For now try to get this plugin to work on your system and experiment some with the code.
